pil_analyzer/src/pil_analyzer.rs

```diff
@@ -454,6 +460,13 @@ impl<T: FieldElement> PILContext<T> {
         assert!(is_new);
     }
 
+    fn handle_assembly(&mut self, _source: SourceRef, asm_statements: Vec<ASMStatement<T>>) {
+        let statements = pilgen::asm_to_pil(asm_statements.into_iter());
```

Comment:
So the assembly blocks are always self contained?

Reply:
What do you mean? They cannot have include statements?

Reply:
yea that they cannot depend on this outside of it

Reply:
What do you mean by depend? Both assembly and pil inside and outside of assembly can freely reference names in all those areas.

Reply:
Right yea, maybe we should clarify the scoping rules in the future.